code/modules/mob/living/living_defense.dm

```diff
@@ -91,44 +91,85 @@
 /mob/living/proc/is_ears_covered()
 	return null
 
-/mob/living/bullet_act(obj/projectile/hitting_projectile, def_zone, piercing_hit = FALSE)
+/mob/living/bullet_act(obj/projectile/proj, def_zone, piercing_hit = FALSE)
 	. = ..()
-	if(. != BULLET_ACT_HIT)
+	var/blocked = check_projectile_armor(def_zone, proj, is_silent = TRUE)
+	if(blocked >= 100)
+		if(. == BULLET_ACT_HIT && proj.is_hostile_projectile())
+			apply_projectile_effects(proj, def_zone, blocked)
 		return .
-	if(!hitting_projectile.is_hostile_projectile())
-		return BULLET_ACT_HIT
 
-	// we need a second, silent armor check to actually know how much to reduce damage taken, as opposed to
-	// on [/atom/proc/bullet_act] where it's just to pass it to the projectile's on_hit().
-	var/armor_check = check_projectile_armor(def_zone, hitting_projectile, is_silent = TRUE)
+	var/hit_limb_zone = check_hit_limb_zone_name(def_zone)
+	var/organ_hit_text = ""
+	if (hit_limb_zone)
```

Comment:
This can be dead code since it's only used after the following condition

In fact I think you may as well return this to the top of the proc as it used to be (in current code) unless there's a reason in particular why that was deleted